plugin/pkg/timeouts/timeouts_test.go

```diff
@@ -0,0 +1,65 @@
+package timeouts
+
+import (
+	"testing"
+	"time"
+)
+
+func TestNewTimeoutFromArg(t *testing.T) {
```

Comment:
I think this would be better as table test
```
func TestNewTimeoutFromArg(t *testing.T) {
	tests := []struct {
		name    string
		arg     string
		wantErr bool
		want    time.Duration
	}{
		{
			name: "valid GO duration - seconds",
			arg:  "30s",
			want: 30 * time.Second,
		},
		{
			name: "valid GO duration - minutes",
			arg:  "2m",
			want: 2 * time.Minute,
		},
		{
			name: "number - fallback to seconds",
			arg:  "30",
			want: 30 * time.Second,
		},
		{
			name:    "invalid timeout",
			arg:     "twenty seconds",
			wantErr: true,
		},
		{
			name:    "timeout is too low",
			arg:     "0",
			wantErr: true,
		},
		{
			name:    "timeout is too high",
			arg:     "24h1s",
			wantErr: true,
		},
	}
	for _, test := range tests {
		t.Run(test.name, func(t *testing.T) {
			actual, err := NewTimeoutFromArg(test.arg)
			if test.wantErr && err == nil {
				t.Error("error was expected")
			}
			if !test.wantErr && err != nil {
				t.Error("error was not expected")
			}

			if test.want != actual {
				t.Errorf("expected '%v' got '%v'", test.want, actual)
			}
		})
	}
}
```